independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java

```diff
@@ -61,11 +61,29 @@ protected boolean addDependency(AppArtifactCoords coords, boolean managed) {
     }
 
     static boolean addDependencyInModel(Model model, AppArtifactCoords coords, boolean managed) {
-        StringBuilder newDependency = new StringBuilder()
-                .append("    implementation '")
-                .append(coords.getGroupId())
-                .append(":")
-                .append(coords.getArtifactId());
+        boolean isBOM = "pom".equals(coords.getType());
+        StringBuilder newDependency;
+        if (isBOM) {
+            // Check if BOM is not included already
+            String resolvedPlatform = String
+                    .format("%s:%s", getProperty(model, "quarkusPlatformGroupId"),
+                            getProperty(model, "quarkusPlatformArtifactId"));
```

Comment:
It's better but works on the assumption that these properties are used. In general there could be any property or direct coordinates.